src/employee/employee.repository.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  EmployeeType,
+  EmployeeCreateType,
+  EmployeeUniqueProperty,
+} from "./interfaces";
+
+export interface IEmployeeRepository {
+  createEmployee: (employee: EmployeeCreateType) => Promise<EmployeeType>;
+  getAllEmployees: () => Promise<EmployeeType[]>;
+  findUnique: (
+    uniqueProperty: EmployeeUniqueProperty,
+    value: number | string
+  ) => Promise<EmployeeType | undefined>;
+}
+
+class EmployeeRepository implements IEmployeeRepository {
+  private counter = 0;
+  private readonly employees: EmployeeType[] = [];
+
+  async createEmployee(employee: EmployeeCreateType) {
+    const newEmployee = { id: this.counter, ...employee };
+    this.employees.push(newEmployee);
+    this.counter++;
```

Comment:
A ten counter nie jest dostarczany z automatu dzięki polu .lenght z employees?

Reply:
teraz sobie myślę, że tak i nie, bo jeżeli się będziemy opierać tylko na liczbie elementów tablicy, to w przypadku, usunięcia jakiegoś elementu, to id mogą się powtórzyć, dlatego imho jednak to nie jest dobry pomysł. A tak czy siak, za chwilę się to usunie, bo wleci bazka danych